### ISSUE TITLE
1.7+ crashes `console.log(window)`

### ISSUE THREAD

**Issue by fatso83** [cjohansen/Sinon.JS#822]
Running `console.log(window)` will throw the following error:
```
TypeError{message: ''undefined' is not an object (evaluating 'this.proxy.toString')', line: 1204, sourceId: 183278944, sourceURL: 'http://localhost:9876/absolute/Users/carl-erik.kopseng/dev-priv/Sinon.JS/node_modules/sinon/pkg/sinon.js?07b1eea46438bf7345a02fa3ddfd7de1b7409eb7', stack: 'TypeError: 'undefined' is not an object (evaluating 'this.proxy.toString')
```

I uploaded an executable test of this on my [console.log.bug](https://github.com/fatso83/Sinon.JS/tree/console.log.bug) branch.

Verifying bug
```
git clone https://github.com/fatso83/Sinon.JS.git
git checkout console.log.bug
npm install
karma start \
        test/console.log_window-bug/karma.conf.js \
        --singleRun
```

Verify it was ok in 1.6.0
```
npm install sinon@1.6.0
karma start \
        test/console.log_window-bug/karma.conf.js \
        --singleRun
```

Tested with PhantomJS 1.9.2

**Comment by fatso83** [comment]
I looked through the commit logs, but I could not see anything specific that would cause this behavior.

**Comment by mroderick** [comment]
That is interesting. Have you tried the same test with Buster.JS?

**Comment by fatso83** [comment]
After googling the error it seems as if this issue is a duplicate of one from January: #648. Seems like you pushed a fix just hours ago (#833).

**Comment by fatso83** [comment]
I have now tried running the same test in Buster.JS, and it seems to work. It might seems as the ones suggesting this might have something to do with Karma or Mocha might be right, in that it seems to interfere somehow with Sinon's magic.

I'll close this one, since it's a duplicate, but I'll put some of the info (like the "offending" commit) I have found in the other issue.
